samples/snippets/csharp/VS_Snippets_CLR/conceptual.asyncdisposable/disposeasync.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+public class ExampleAsyncDisposable : IAsyncDisposable, IDisposable
+{
+    // To detect redundant calls
+    private bool _disposed = false;
+
+    // Created in .ctor, omitted for brevity.
+    private Utf8JsonWriter _jsonWriter;
+
+    public async ValueTask DisposeAsync()
+    {
+        await DisposeAsyncCore();
+
+        Dispose(false);
+        GC.SuppressFinalize(this);
+    }
+
+    protected virtual async ValueTask DisposeAsyncCore()
+    {
+        // Cascade async dispose calls
+        if (_jsonWriter != null)
+        {
+            await _jsonWriter.DisposeAsync();
+        }
+    }
+
+    public void Dispose()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+
+    protected virtual void Dispose(bool disposing)
+    {
+        if (_disposed)
+        {
+            return;
+        }
+
+        if (disposing)
+        {
+            // TODO: dispose managed state (managed objects).
```

Comment:
I would think that `_jsonWriter` would need to be synchronously disposed of here, correct?

```suggestion
            _jsonWriter?.Dispose();
            // TODO: dispose managed state (managed objects).
```